src/Promact.TestCaseManagement.Core/Controllers/ModuleController.cs

```diff
@@ -0,0 +1,156 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Promact.TestCaseManagement.DomainModel.Models;
+using Promact.TestCaseManagement.Repository.ApplicationClass.Module;
+using Promact.TestCaseManagement.Repository.ModuleRepository;
+using Promact.TestCaseManagement.Repository.ProjectRepository;
+using Promact.TestCaseManagement.Utility.Constants;
+using System.Threading.Tasks;
+
+namespace Promact.TestCaseManagement.Core.Controllers
+{
+    [Route(StringConstants.ProjectBaseUrl)] 
+    public class ModuleController : Controller
+    {
+        #region "Private Member(s)"
+        readonly IModuleRepository _iModuleRepository;
+        readonly IProjectRepository _iProjectRepository;
+        #endregion
+
+        #region "Constructor"
+        public ModuleController(IModuleRepository iModuleRepository, IProjectRepository iProjectRepository)
+        {
+            _iModuleRepository = iModuleRepository;
+            _iProjectRepository = iProjectRepository;
+        }
+        #endregion
+
+        #region "Public API(s)"
+
+        /// <summary>
+        /// Method to get all modules
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <returns></returns>
+        [HttpGet("{projectId}/module")]
+        public async Task<IActionResult> GetModulesAsync(int projectId)
+        {
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+
+            var listOfModules = await _iModuleRepository.GetModulesAsync(projectId);
+
+            return Ok(listOfModules);
```

Comment:
please add await _iModuleRepository.GetModulesAsync(projectId); call into Ok method. One variable space can be optimized.